sdk/include/opentelemetry/sdk/logs/logger.h

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "opentelemetry/logs/log_record.h"
+#include "opentelemetry/logs/logger.h"
+#include "opentelemetry/nostd/shared_ptr.h"
+#include "opentelemetry/nostd/string_view.h"
+#include "opentelemetry/sdk/common/atomic_shared_ptr.h"
+#include "opentelemetry/sdk/logs/logger_provider.h"
+#include "opentelemetry/sdk/logs/processor.h"
+
+#include <unordered_map>
+#include <vector>
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace sdk
+{
+namespace logs
+{
+class Logger final : public opentelemetry::logs::Logger, public std::enable_shared_from_this<Logger>
+{
+public:
+  /**
+   * Initialize a new logger and pass it a processor.
+   * @param processor The processor for this logger.
+   */
+  explicit Logger(std::shared_ptr<opentelemetry::logs::LoggerProvider> lp) noexcept;
+
+  /**
+   * Writes a log record into the processor. The log record will only be
+   * sent if the severity level is greater than the minimum severity set by
+   * SetMinSeverity(s).
+   * @param record The record to write into the processor.
+   */
+  void log(const opentelemetry::logs::LogRecord &record) noexcept override;
+
+private:
+  // The minimum severity of this Logger
+  opentelemetry::logs::Severity minSeverity_;
```

Comment:
Not needed?

Reply:
No? The logger constructor currently sets the minSeverity and Logger::log() checks if the logrecord passed in meets the minSeverit or not. Should this be part of a future PR?

Reply:
Yes, I think this should be part of the sampler design/PR.

Reply:
Two things:

1. I agree this portion of the API should be deferred and discussed.
2. There (can be) a significant performance boost to filtering out logs at the Logger level.   Once we have a sampler design/PR *and* benchmark tests, we should re-evaluate minSeverity-in-logger.